x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_validator/util.ts

```diff
@@ -0,0 +1,161 @@
+    const msg = i18n.translate('xpack.ml.newJob.wizard.validateJob.jobNameAlreadyExists', {
+      defaultMessage:
+        'Job ID already exists. A job ID cannot be the same as an existing job or group.',
+    });
+    basicValidations.jobId.message = msg;
+  }
+
+  if (validationResults.contains('job_group_id_invalid')) {
+    basicValidations.groupIds.valid = false;
+    const msg = i18n.translate(
+      'xpack.ml.newJob.wizard.validateJob.jobGroupAllowedCharactersDescription',
+      {
+        defaultMessage:
+          'Job group names can contain lowercase alphanumeric (a-z and 0-9), hyphens or underscores; ' +
+          'must start and end with an alphanumeric character',
+      }
+    );
+    basicValidations.groupIds.message = msg;
+  } else if (validationResults.contains('job_group_id_already_exists')) {
+    basicValidations.groupIds.valid = false;
+    const msg = i18n.translate('xpack.ml.newJob.wizard.validateJob.groupNameAlreadyExists', {
+      defaultMessage:
+        'Group ID already exists. A group ID cannot be the same as an existing job or group.',
+    });
+    basicValidations.groupIds.message = msg;
+  }
+
+  if (validationResults.contains('model_memory_limit_units_invalid')) {
+    basicValidations.modelMemoryLimit.valid = false;
+    const str = `${ALLOWED_DATA_UNITS.slice(0, ALLOWED_DATA_UNITS.length - 1).join(', ')} or ${[
+      ...ALLOWED_DATA_UNITS,
+    ].pop()}`;
+    const msg = i18n.translate(
+      'xpack.ml.newJob.wizard.validateJob.modelMemoryLimitUnitsInvalidErrorMessage',
+      {
+        defaultMessage: 'Model memory limit data unit unrecognized. It must be {str}',
+        values: { str },
+      }
+    );
+    basicValidations.modelMemoryLimit.message = msg;
+  }
+
+  if (validationResults.contains('model_memory_limit_invalid')) {
+    basicValidations.modelMemoryLimit.valid = false;
+    const msg = i18n.translate(
+      'xpack.ml.newJob.wizard.validateJob.modelMemoryLimitRangeInvalidErrorMessage',
+      {
+        defaultMessage:
+          'Model memory limit cannot be higher than the maximum value of {maxModelMemoryLimit}',
+        values: { maxModelMemoryLimit: limits.max_model_memory_limit.toUpperCase() },
+      }
+    );
+    basicValidations.modelMemoryLimit.message = msg;
+  }
+
+  if (validationResults.contains('detectors_duplicates')) {
+    basicValidations.duplicateDetectors.valid = false;
+    const msg = i18n.translate(
+      'xpack.ml.newJob.wizard.validateJob.duplicatedDetectorsErrorMessage',
+      {
+        defaultMessage: 'Duplicate detectors were found.',
+      }
+    );
+    basicValidations.duplicateDetectors.message = msg;
+  }
+
+  if (validationResults.contains('bucket_span_empty')) {
+    basicValidations.bucketSpan.valid = false;
+    const msg = i18n.translate(
+      'xpack.ml.newJob.wizard.validateJob.bucketSpanMustBeSetErrorMessage',
+      {
+        defaultMessage: '{bucketSpan} must be set',
+        values: { bucketSpan: 'bucket_span' },
```

Comment:
I think this looks better without the underscore as the user doesn't need to know the underlying field name, and should be `Bucket span` instead